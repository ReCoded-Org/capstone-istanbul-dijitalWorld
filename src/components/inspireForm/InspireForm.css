.inspireForm {
  margin: 5%;
}

.inspireFirstRow {
  height: 10%;
  font-size: 2rem;
}

.inspireInputBox::-webkit-input-placeholder {
  font-size: 1.25rem;
}

.inspireButton {
  background-color: #6c567b;
  color: white;
  cursor: pointer;
  width: 13.75rem;
  height: 3.25rem;
  font-size: 1.5rem;
  border-radius: 3.125rem;
  border-style: hidden;
  box-shadow: 0 0.6875rem 1.5rem -0.4375rem #b3b3b3;
}

.inspireButton:hover {
  background-color: #f8b195;
}

.inspireButtonRow {
  display: flex;
  justify-content: center;
}

.inspireInputBox {
  border-color: #f8b195;
  background-color: #f4f4f4;
}
