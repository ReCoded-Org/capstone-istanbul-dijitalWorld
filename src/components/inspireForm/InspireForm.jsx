import React, { useState } from 'react';
import './InspireForm.css';
import database from '../../firebase';
import { Form, Col, Row, Alert } from 'react-bootstrap';
import { useTranslation } from 'react-i18next';

export default function InspireForm() {
  const { t } = useTranslation();
  const [headline, setHeadline] = useState();
  const [category, setCategory] = useState();
  const [story, setStory] = useState();
  const [show, setShow] = useState(false);

  const handleHeadlineInput = async (e) => {
    setHeadline(e.target.value);
  };

  const handleCategoryInput = async (e) => {
    setCategory(e.target.value);
  };

  const handleStoryInput = async (e) => {
    setStory(e.target.value);
  };
  function ThankYouPopUp() {
  
    return (
      <>
        <Alert show={show} variant="success">
          <Alert.Heading>Thank You!</Alert.Heading>
          <p>
           Thank you for submitting your story and having the courage to share!<br/>
           Our admins will review the story and help you get it out there as soon as possible!
          
          </p>
          <hr />
        </Alert>
        </>
    );
  }  

  const handleSubmitStory = (e) => {
    e.preventDefault();

    const newStory= database.collection('inspire').doc()
    newStory.set({
      headline: headline,
      category: category,
      story: story,
    });
    setShow(true)
  };
  
  return (
    <Form className="inspireForm" onSubmit={handleSubmitStory}>
      <Row className="inspireFirstRow">
        <Col>
          <Form.Group>
            <Form.Control
              size="lg"
              type="input"
              placeholder={t('inspire.subject')}
              className="inspireInputBox"
              onChange={handleHeadlineInput}
              required
            />
          </Form.Group>
        </Col>
        <Col>
          <Form.Group>
            <Form.Control size="lg" as="select" className="inspireInputBox" onChange={handleCategoryInput}required>
              <option default>{t('inspire.category.0')}</option>
              <option>{t('inspire.category.1')}</option>
              <option>{t('inspire.category.2')}</option>
              <option>{t('inspire.category.3')}</option>
              <option>{t('inspire.category.4')}</option>
              <option>{t('inspire.category.5')}</option>
            </Form.Control>
          </Form.Group>
        </Col>
      </Row>
      <Form.Group>
        <Form.Control
          as="textarea"
          rows="15"
          placeholder={t('inspire.story')}
          className="inspireInputBox"
          onChange={handleStoryInput}
          required
        />
      </Form.Group>
      <Row><ThankYouPopUp/></Row>
      <Row className="inspireButtonRow">
        <button type="submit" className="inspireButton" >
          Inspire
        </button>
      </Row>
    </Form>
  );
}
