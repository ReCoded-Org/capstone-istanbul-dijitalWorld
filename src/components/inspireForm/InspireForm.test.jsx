import React from 'react';
import InspireForm from './InspireForm';
import renderer from 'react-test-renderer';

it('renders correctly', () => {
  const tree = renderer.create(<InspireForm />).toJSON();
  expect(tree).toMatchSnapshot();
});
