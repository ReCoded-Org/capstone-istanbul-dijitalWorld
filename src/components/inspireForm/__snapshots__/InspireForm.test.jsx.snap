// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<form
  className="inspireForm"
  onSubmit={[Function]}
>
  <div
    className="inspireFirstRow row"
  >
    <div
      className="col"
    >
      <div
        className="form-group"
      >
        <input
          className="inspireInputBox form-control form-control-lg"
          onChange={[Function]}
          placeholder="inspire.subject"
          required={true}
          type="input"
        />
      </div>
    </div>
    <div
      className="col"
    >
      <div
        className="form-group"
      >
        <select
          className="inspireInputBox form-control form-control-lg"
          onChange={[Function]}
          required={true}
        >
          <option
            default={true}
          >
            inspire.category.0
          </option>
          <option>
            inspire.category.1
          </option>
          <option>
            inspire.category.2
          </option>
          <option>
            inspire.category.3
          </option>
          <option>
            inspire.category.4
          </option>
          <option>
            inspire.category.5
          </option>
        </select>
      </div>
    </div>
  </div>
  <div
    className="form-group"
  >
    <textarea
      className="inspireInputBox form-control"
      onChange={[Function]}
      placeholder="inspire.story"
      required={true}
      rows="15"
    />
  </div>
  <div
    className="row"
  />
  <div
    className="inspireButtonRow row"
  >
    <button
      className="inspireButton"
      type="submit"
    >
      inspire.button
    </button>
  </div>
</form>
`;
